### PR TITLE
add NegativeFloatType scalar

### PR DESCRIPTION
**Pull request by wonbyte**
Add `NegativeFloatType` Scalar
- Addresses #2968

### CODE REVIEW

File: src/HotChocolate/Core/src/Types.Scalars/NegativeFloatType.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        protected override double ParseLiteral(IFloatValueLiteral valueSyntax)
+        {
+            if (valueSyntax.ToDouble() >= MaxValue)
+            {
+                throw ThrowHelper.NegativeFloatType_ParseLiteral_IsNotNegative(this);
+            }
+
+            return base.ParseLiteral(valueSyntax);
+        }
+
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Types.Scalars/NegativeFloatType.cs:52]: Can we also add `/// <inheritdoc />`

File: src/HotChocolate/Core/src/Types.Scalars/ThrowHelper.cs [hunk 2]
```diff
@@ -24,6 +24,28 @@ public static SerializationException EmailAddressType_ParseValue_IsInvalid(IType
                 type);
         }
 
+        public static SerializationException NegativeFloatType_ParseLiteral_IsNotNegative(IType type)
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Types.Scalars/ThrowHelper.cs:27]: We have a maximum policy of 100 characters per line :)

File: website/src/docs/hotchocolate/defining-a-schema/scalars.md [hunk 3]
```diff
@@ -410,3 +410,11 @@ The PhoneNumber scalar type represents a value that conforms to the standard E.1
 """
 scalar PhoneNumber
 ```
+
+## NegativeFloat
+```sdl
+"""
+The NegativeFloat scalar type represents a signed real number with a value less than 0
```
Comment (PascalSenn) [inline website/src/docs/hotchocolate/defining-a-schema/scalars.md:417]: This comment does not look right :)

File: src/HotChocolate/Core/test/Types.Scalars.Tests/NegativeFloatTypeTests.cs [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+            // act
+            // assert
+            schema.ToString().MatchSnapshot();
+        }
+
+        [Theory]
+        [InlineData(typeof(EnumValueNode), TestEnum.Foo, false)]
+        [InlineData(typeof(FloatValueNode), 1d, false)]
+        [InlineData(typeof(FloatValueNode), -1.0d, true)]
```
Comment (PascalSenn) [inline src/HotChocolate/Core/test/Types.Scalars.Tests/NegativeFloatTypeTests.cs:24]: cann we also add
```
[InlineData(typeof(FloatValueNode), 0d, false)]
```

File: src/HotChocolate/Core/src/Types/Types/Scalars/FloatTypeBase.cs [hunk 5]
```diff
@@ -53,6 +53,22 @@ public override bool IsInstanceOfType(IValueNode valueSyntax)
...
+                return true;
+            }
+
+            if (runtimeValue is TRuntimeType t)
+            {
+                return IsInstanceOfType(t);
+            }
+
+            return false;
+        }
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Types/Types/Scalars/FloatTypeBase.cs:70]: @michaelstaib  @wonbyte  fixed the base class, could this infer with something?

**Review by PascalSenn — CHANGES_REQUESTED** [review]
Awesome! 
They really get better every time
just a couple of things :)

**Comment by wonbyte** [comment]
> Awesome!
> They really get better every time
> just a couple of things :)

I'll eventually get it to 0........

**Review by PascalSenn — APPROVED** [review]

**Pull request merged**